### PR TITLE
Salesforce : Expand producer test coverage

### PR DESCRIPTION
**Pull request by zbendhiba**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-tests/salesforce/src/test/java/org/apache/camel/quarkus/component/salesforce/SalesforceTest.java [hunk 1]
```diff
@@ -81,4 +102,119 @@ public void testBulkJobApi() {
...
+            RestAssured.get("/salesforce/account/" + accountId)
+                    .then()
+                    .statusCode(200)
+                    .body(
+                            "Id", not(emptyString()),
+                            "AccountNumber", not(emptyString()));
+
+            // delete the account
+            // Clean up
+            RestAssured.delete("/salesforce/account/" + accountId)
```
Comment (aldettinger) [inline integration-tests/salesforce/src/test/java/org/apache/camel/quarkus/component/salesforce/SalesforceTest.java:184]: Would the delete not be invoked due to crash/exception, I think subsequent tests should be fine as the accountId would not be the same.

Comment (zbendhiba) [reply integration-tests/salesforce/src/test/java/org/apache/camel/quarkus/component/salesforce/SalesforceTest.java:184]: yes, thanks @aldettinger. I'd fix that

Comment (zbendhiba) [reply integration-tests/salesforce/src/test/java/org/apache/camel/quarkus/component/salesforce/SalesforceTest.java]: The latest commit addresses this unnecessary not null test on the accountId. Thanks for catching this

Comment (aldettinger) [reply integration-tests/salesforce/src/test/java/org/apache/camel/quarkus/component/salesforce/SalesforceTest.java]: @zbendhiba I was more thinking about cases like the "get" before the "delete" that would throw a network exception. And, in this rare case, there should only be an account leak and subsequent tests should be fine.

I think, it should be fine. CI is only running mock tests and worst case one would need to cleanup the salesforce account from time to time on manual test.

**Review by aldettinger — APPROVED** [review]

**Review by ppalaga — APPROVED** [review]

**Pull request merged**